api/v1beta1/azuremachine_webhook.go

```diff
@@ -143,6 +143,13 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 		allErrs = append(allErrs, err)
 	}
 
+	if err := webhookutils.ValidateImmutable(
```

Comment:
where is the defaulting happening for AzureMachine?

Reply:
Added defaulting and validation + testing for AzureMachine.